### PR TITLE
refactor(urql): infinite pending

### PR DESCRIPTION
**Pull request by douglaskariuki**

### CODE REVIEW

**Review by dai-shi — COMMENTED** [review]
Thanks for working on this. I can merge this if tests are passing.
I'm also working on a full re-implementation, so this will be only short lived even if it's merged.

File: src/urql/atomWithMutation.ts [hunk 1]
```diff
@@ -19,17 +19,33 @@ export function atomWithMutation<Data, Variables extends AnyVariables>(
...
+  const operationResultAtom = atom((get) => {
+    let resolve: ((result: Result) => void) | null = null
+    const makePending = () => 
+      new Promise<Result>((r) => {
+        resolve = r
+      })
+
+    const resultAtom = atom<Result | Promise<Result>>(makePending())
+    return { resultAtom, makePending }
+  })
```
Comment (dai-shi) [inline src/urql/atomWithMutation.ts:32]: On second look, this looks wrong. Do you plan to work on this refactor and fix?

Comment (douglaskariuki) [reply src/urql/atomWithMutation.ts:32]: I wanted to fix infinite pending in operationResultAtom by calling the pending function when there's no result

Comment (douglaskariuki) [reply src/urql/atomWithMutation.ts:32]: I'm also supposed to add typescript types to the values returned in operationResultAtom

Comment (dai-shi) [reply src/urql/atomWithMutation.ts:32]: I thought this was a refactoring, but do you have any issues? In such a case, you want to reproduce the issue with test. That would  be very valuable.

Comment (douglaskariuki) [reply src/urql/atomWithMutation.ts:32]: I will update test file and update this PR once all tests pass

**Comment by dai-shi** [comment]
I'm still interested in reproducing the issue in tests.
For now, let me close this as the base implementation is now in `jotai-urql`.

**Comment by douglaskariuki** [comment]
I'll be following up on the jotai-urql framework too and update this main one where needed or when all tests are successful

**Pull request closed without merging**